app/src/main/java/com/example/composesigninsample/MainActivity.kt

```diff
@@ -89,12 +124,27 @@ fun SignInScreen(loggedOutEvent: () -> Unit, viewModel: SignInViewModel = hiltVi
                 label = { Text("Password") },
                 singleLine = true
             )
+
+            Spacer(modifier = Modifier.height(24.dp))
+
             Text(text = "Welcome to the app. Sign in to begin")
-            Button(onClick = viewModel::signIn) {
+
+            Spacer(modifier = Modifier.height(8.dp))
+
+            Button(onClick = setLoggedIn) {
                 Text(text = "Sign In")
             }
         }
     }
+
+    DisposableEffect(key1 = isSignedIn, effect = {
+        if (isSignedIn) {
+            navigateToHome()
+        }
+
+        onDispose {
+        }
+    })
```

Comment:
I like this since I don't need an if statement for `isSignedIn` and instead basically declare the effect with it's key.

I am new to LaunchedEffect and DisposableEffect though so I will have to read up the differences on why you chose to replace LaunchedEffect with Disposable.

Reply:
I replaced it with DisposableEffect because you don't actually need the coroutine that is executed with LaunchedEffect.